### ISSUE TITLE
genie_python: Tidy up API

### ISSUE THREAD

**Issue by DominicOram** [ISISComputingGroup/IBEX#5457]
As a user I would like the genie_python API to be as small as possible so that it's easier to find the functions I use a lot. After a meeting on 04/06/2020 with @FreddieAkeroyd, @DominicOram, @ThomasLohnert, @Tom-Willemsen, @John-Holt-Tessella, @davidkeymer, @JamesKingWork, @RaiBishal and @Alexandru-Dascalu the following acceptance criteria were created:

## Acceptance Criteria
- [x] `check_lowlimit_against_highlimit` - should be moved to be an internal function
- [x] `set_title`, `set_script_dir`, `set_number_soft_periods`, `set_sample_par`, `set_beamline_par`, `set_period` - remove as these have been marked as depreciated for ~3 years. They directly map to `change_x` equivalents
- [x] `get_absolute_path`, `is_absolute` - should be moved to be an internal function
- [x] `get_instrument_py_name` - check if used, specifically on emma-a and in scans library. If used move into advanced, if not remove
- [x] `get_instrument` - move into advanced
- [x] `send_tcpip` - move into shared instrument scripts, confirm that it works in python 3
- [x] `set_messages_verbosity` - move to advanced and rename to make it clear that this is for DAE verbosity

- [x] For all of the above search instrument scripts for usage, where used contact specific instrument scientists
- [x] Ensure that the release notes detail these changes well and circulate as part of https://github.com/ISISComputingGroup/IBEX/issues/5389

**Comment by DominicOram** [comment]
Used https://github.com/ISISComputingGroup/ConfigChecker/pull/25 to find that:
* `set_title` used by MERLIN ([new config branch with fix](https://control-svcs.isis.cclrc.ac.uk/git/?p=instconfigs/inst.git;a=shortlog;h=refs/heads/NDXMERLIN_update_genie))
* `set_title` used on LARMOR ([new config branch with fix](https://control-svcs.isis.cclrc.ac.uk/git/?p=instconfigs/inst.git;a=shortlog;h=refs/heads/NDXLARMOR_update_genie))

Ran a similar script on InstrumentScripts and found no usages.

Used a script that is now in `ISIS_Experiment_Controls\private_git_repositories\private_scripts.git` to find the following:
* IMAT were using `set_period` and `set_title` in 2016, haven't since
* ZOOM used `set_script_dir` in late 2019
* VESUVIO were using `set_title` in 2017, haven't since
* EMMA used `set_title` extensively in 2019
* CRISP used `set_period` and `set_script_dir` once in 2019

**Comment by DominicOram** [comment]
I propose sending the following email to scientists on MERLIN, LARMOR, ZOOM, EMMA and CRISP, once this has been merged:

Hi All,

One of the common complaints we get about genie_python is the large list of functions available when trying to use autocomplete in the GUI. With the Python 3 migration giving everyone an opportunity to look back over their scripts we thought that now would be a good time to move some of the functions that are not widely used so that they do not confuse users. 

Since 2017 every time a user called set_title, set_script_dir and set_period a message has been displayed encouraging people to instead use change_title, change_script_dir or change_period respectively. To avoid this duplication we are now removing the set_ commands and will now only have the change_ commands.

MERLIN and LARMOR make use of set_title in a couple of instrument scripts, we have changed these for you. From the logs it looks like ZOOM, EMMA and CRISP have been using these functions a bit last year so we wanted to make you specifically aware. We will also be letting all scientists know about these changes when we update the IBEX client to Python 3 in the next couple of weeks.

We do not believe any of the other changes we've made will effect any instruments but if you are interested the full list is at https://github.com/ISISComputingGroup/IBEX/issues/5457.

Thanks,
Dom
